test/test_gaussian_datatypes.py

```diff
@@ -0,0 +1,443 @@
+      0.496955043655 -0.852791790900 -0.095944016600 -0.454266086000  0.380324658600  0.893564918400
+      0.162491615040 -0.242351537800  1.102830348700 -0.257388983000  1.054102919900  0.152954188700
+
+ O  MY-TZVP-MOLOPT-GTH MY-TZVP-MOLOPT-GTH-q6
+ 1
+ 2 0 2 7 3 3 1
+     12.015954705512 -0.060190841200  0.065738617900  0.041006765400  0.036543638800 -0.034210557400 -0.000592640200  0.014807054400
+      5.108150287385 -0.129597923300  0.110885902200  0.080644802300  0.120927648700 -0.120619770900  0.009852349400  0.068186159300
+      2.048398039874  0.118175889400 -0.053732406400 -0.067639801700  0.251093670300 -0.213719464600  0.001286509800  0.290576499200
+      0.832381575582  0.462964485000 -0.572670666200 -0.435078312800  0.352639910300 -0.473674858400 -0.021872639500  1.063344189500
+      0.352316246455  0.450353782600  0.186760006700  0.722792798300  0.294708645200  0.484848376400  0.530504764700  0.307656114200
+      0.142977330880  0.092715833600  0.387201458600 -0.521378340700  0.173039869300  0.717465919700 -0.436184043700  0.318346834400
+      0.046760918300 -0.000255945800  0.003825849600  0.175643142900  0.009726110600  0.032498979400  0.073329259500 -0.005771736600
+"""
+
+
+@pytest.mark.process_execution
+def test_gdts_validation(cp2k_code, clear_database):  # pylint: disable=unused-argument
+    """Testing CP2K with the Basis Set stored in gaussian.basisset"""
+
+    import ase.build
+
+    from aiida.engine import run_get_node
+    from aiida.plugins import CalculationFactory
+    from aiida.orm import Dict, StructureData
+
+    # structure
+    atoms = ase.build.molecule("H2O")
+    atoms.center(vacuum=2.0)
+    structure = StructureData(ase=atoms)
+
+    fhandle_bset = StringIO(BSET_INPUT)
+    fhandle_pseudo = StringIO(PSEUDO_INPUT)
+    bsets = {b.element: b for b in BasisSet.from_cp2k(fhandle_bset)}
+    pseudos = {p.element: p for p in Pseudo.from_cp2k(fhandle_pseudo)}
+
+    # parameters
+    parameters = Dict(
+        dict={
+            "FORCE_EVAL": {
+                "METHOD": "Quickstep",
+                "DFT": {
+                    "QS": {
+                        "EPS_DEFAULT": 1.0e-12,
+                        "WF_INTERPOLATION": "ps",
+                        "EXTRAPOLATION_ORDER": 3,
+                    },
+                    "MGRID": {
+                        "NGRIDS": 4,
+                        "CUTOFF": 280,
+                        "REL_CUTOFF": 30
+                    },
+                    "XC": {
+                        "XC_FUNCTIONAL": {
+                            "_": "LDA"
+                        }
+                    },
+                    "POISSON": {
+                        "PERIODIC": "none",
+                        "PSOLVER": "MT"
+                    },
+                },
+                "SUBSYS": {
+                    "KIND": [
+                        {
+                            "_": "O",
+                            "POTENTIAL": "GTH " + pseudos["O"].name,
```

Comment:
Yes, maybe the newly added examples are a bit clearer in that regard.
Ideally people can omit the KIND section completely and only select basissets and pseudos by providing the respective list. But CP2K allows to mix different basissets in multiple ways:
* different sets for same kind but different sites
* multiple sets for same kind and same site (in which case CP2K merges the specified basissets)
* multiple FORCE_EVAL

Hence we need a way to allow the user to explicitly specify the assignment and since we already have one, we can simply reuse it.

The "GTH" (and "ORB") are just new-style CP2K options to explicitly specify the purpose/format.